### PR TITLE
[ML][Inference] Adding preprocessors to definition object

### PR DESCRIPTION
**Pull request by benwtrent**
Within the definition of a trained model, we should include the pre-processing that needs to occur based on the training data.

### CODE REVIEW

File: client/rest-high-level/src/main/java/org/elasticsearch/client/ml/inference/TrainedModelDefinition.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            true,
+            TrainedModelDefinition.Builder::new);
+    static {
+        PARSER.declareNamedObjects(TrainedModelDefinition.Builder::setTrainedModel,
+            (p, c, n) -> p.namedObject(TrainedModel.class, n, null),
+            (modelDocBuilder) -> { /* Noop does not matter client side*/ },
+            TRAINED_MODEL);
+        PARSER.declareNamedObjects(TrainedModelDefinition.Builder::setPreProcessors,
+            (p, c, n) -> p.namedObject(PreProcessor.class, n, null),
+            (trainedModelDefBuilder) -> {/* Does not matter client side*/ },
```
Comment (przemekwitek) [inline client/rest-high-level/src/main/java/org/elasticsearch/client/ml/inference/TrainedModelDefinition.java]: nit: Could you make comments in lines 48 and 52 equal?

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java [hunk 2]
```diff
@@ -242,6 +230,9 @@ public int hashCode() {
     }
 
 
+    public static class Definition {
```
Comment (przemekwitek) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java]: This is not needed, is it?

**Review by przemekwitek — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java (hunk 2 above)
Comment (benwtrent) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfig.java]: buahaha :D nope!

**Pull request merged**